libminifi/src/utils/ChecksumCalculator.cpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include "utils/ChecksumCalculator.h"
+
+#include <fstream>
+
+#include "sodium/crypto_hash_sha256.h"
+#include "utils/file/FileUtils.h"
+#include "utils/StringUtils.h"
+
+namespace {
+
+const std::string AGENT_IDENTIFIER_KEY = "nifi.c2.agent.identifier=";
+
+}
+
+namespace org {
+namespace apache {
+namespace nifi {
+namespace minifi {
+namespace utils {
+
+void ChecksumCalculator::setFileLocation(const std::string& file_location) {
+  file_location_ = file_location;
+  file_name_ = utils::file::get_child_path(file_location);
```

Comment:
should we invalidate the checksum here?

Reply:
Yes, we do, good catch, thanks!  (We don't currently call `setFileLocation()` twice on the same object, but someone may do that in the future.)  Fixed in https://github.com/apache/nifi-minifi-cpp/pull/1123/commits/7369c4998a7b56ac88a38a2b9dc4ded508193dff.